src/api/functions/resolve-symlink.ts

```diff
@@ -52,5 +87,9 @@ export function build(
 ): ResolveSymlinkFunction | null {
   if (!options.resolveSymlinks) return null;
 
-  return isSynchronous ? resolveSymlinksSync : resolveSymlinksAsync;
+  if (isSynchronous) {
+    return options.useRealPaths ? resolveSymlinksWithRealPathsSync :  resolveSymlinksSync
+  }
+
+  return options.useRealPaths ? resolveSymlinksWithRealPathsAsync : resolveSymlinksAsync;
```

Comment:
Let's invert these checks to be more consistent.

```suggestion
  if (options.useRealPaths)
    return isSynchronous
        ? resolveSymlinksWithRealPathsSync
        : resolveSymlinksWithRealPathsAsync;
  return isSynchronous
    ? resolveSymlinksSync
    : resolveSymlinksAsync;
```